### PR TITLE
ES2015ify

### PR DESCRIPTION
**Pull request by SamVerschueren**
I'd suggest to first release a minor and then merge this one and release a breaking change.

### CODE REVIEW

File: lenient.js [hunk 1]
```diff
@@ -59,10 +59,10 @@ var oMatch = {
 };
```
Comment (sindresorhus) [inline lenient.js]: Can you make these use `Map()` instead? Then we don't need all those ugly `({}.hasOwnProperty.call` calls.

File: index.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
+const lenient = require('./lenient');
 
-module.exports = function (val, opts) {
+module.exports = (val, opts) => {
 	val = String(val).trim();
-	opts = objectAssign({
+	opts = Object.assign({
 		lenient: false,
 		default: null
 	}, opts);
```
Comment (sindresorhus) [inline index.js]: Use template literal in the below typecheck?

Comment (SamVerschueren) [inline index.js]: Definitely!

**Comment by SamVerschueren** [comment]
Sorry for the late reply, was quite busy. Used a `Map` now.

**Comment by sindresorhus** [comment]
No worries. It was not important. Thank you!

**Pull request merged**